libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -479,25 +480,42 @@ uint64_t VoteManager::roundDeterminedFromVotes(size_t two_t_plus_one) {
 }
 
 bool VoteManager::addRewardVote(const std::shared_ptr<Vote>& vote) {
-  if (!verifyRewardVote(vote)) return false;
-
   const auto vote_hash = vote->getHash();
   const auto voted_block_hash = vote->getBlockHash();
+  const auto vote_step = vote->getStep();
+  const auto vote_type = vote->getType();
+
+  if (vote_type != cert_vote_type) {
+    LOG(log_dg_) << "Drop vote " << vote_hash << ". The vote type " << vote_type << " is not cert vote type.";
+    return false;
+  }
+
+  if (vote_step != 3) {
```

Comment:
Why is it should be 3? 
I see that it was this way before you changes, but I think we need to improve checks like this somehow. At least with a comment

Reply:
I dunno why @gaoqi8649  should answer that :)

Reply:
https://github.com/Taraxa-project/taraxa-node/issues/1804